cogs/repmod_moodle.py

```diff
@@ -245,13 +245,25 @@ def file_report(self, usr, grade, comment):
         messages.append(msg)
         logger.info(self._format_msg(msg))
 
-        # Check Moodle User
-        if usr['auth'] != 'moodle':
-            msg = "repmod_moodle: Requires user with authmod 'moodle'"
+        # Check Moodle or LDAP User
+        if usr['auth'] != 'moodle' and usr['auth'] != 'ldap':
+            msg = "repmod_moodle: Requires user with authmod 'moodle' or 'ldap'"
             logger.error(self._format_msg(msg))
             raise MoodleReporterError(msg)
 
         # Extract Vars
+        result = self.ws.core_user_get_users([('username', usr['username'])])
+        moodle_usr_list = result['users']
+        if len(moodle_usr_list) < 1:
+            msg = "repmod_moodle: Username not found"
+            logger.error(self._format_msg(msg))
+            raise MoodleReporterError(msg)
+        elif len(moodle_usr_list) > 1:
+            msg = "repmod_moodle: multiple users found"
+            logger.error(self._format_msg(msg))
+            raise MoodleReporterError(msg)
+        moodle_usr = moodle_usr_list[0]
+        usr['moodle_id'] = moodle_usr['id']
```

Comment:
Is there a reason to set the use['moodle_id'] here? I don;t think we ever use it. Why not just set `user_id = int(moodle_use['id'])`?